blog/content/post/2024-01-31-drm-bridge-chaining.md

```diff
@@ -0,0 +1,863 @@
+        pinctrl_lcd_5v_enable: lcd_5v_en {
+            fsl,pins = <
+                MX8MM_IOMUXC_SPDIF_EXT_CLK_GPIO5_IO5 0x19
+            >;
+        };
+        pinctrl_it6263_en: it6263_en {
+            fsl,pins = <
+                MX8MM_IOMUXC_SAI3_MCLK_GPIO5_IO2 0x19
+            >;
+        };
+        pinctrl_i2c1_sn65dsi84: i2c1-sn65dsi83grp {
+            fsl,pins = <
+                MX8MM_IOMUXC_GPIO1_IO05_GPIO1_IO5 0x04
+                MX8MM_IOMUXC_GPIO1_IO06_GPIO1_IO6 0x106
+            >;
+        };
+    };
+    ```
+
+After above changes both chips were detected on I2C buses, here is `sn65dsi84`
+on `i2c-0` under address `2c`:
+
+```shell
+# i2cdetect -y 0
+     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
+00:                         UU -- -- -- -- -- -- --
+10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+20: -- -- -- -- -- -- -- -- -- -- -- -- 2c -- -- --
+30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+40: -- -- -- -- -- -- -- -- 48 -- -- -- -- -- -- --
+50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+70: -- -- -- -- -- -- -- --
+```
+
+#### Drivers
+
+Then, it was high time to add appropriate drivers to the image. In theory, after
+adding drivers, the entire graphic chain presented in [Structure](#structure)
+chapter above should work, with no need for additional kernel or Weston
+configuration. Let's see then.
+
+Following drivers were chosen to drive this chain:
+
+* for `sn65dsi84`: driver from [`varigit/linux-imx` repository][sn65dsi84-varigit-repo]
+* for `it6263`: driver from [`nxp-imx/linux-imx` repository][it6263-nxp-repo]
+
+[it6263-nxp-repo]: https://github.com/nxp-imx/linux-imx/blob/lf-5.15.y/drivers/gpu/drm/bridge/it6263.c
+
+#### Bridge chaining
+
+The first boot result was not very pleasurable: several errors in `dmesg` from
+`sn65dsi84` driver and no messages from `it6263` driver. No video output was
+noticed as well:
+
+```shell
+# dmesg | grep it6263
+# dmesg | grep sn65dsi84
+[    3.575954] sn65dsi84 0-002c: even_odd_swap property not found, using default
+[    3.583172] sn65dsi84 0-002c: lvds-channels property not found, using default
+[    3.590351] /soc@0/bus@30800000/i2c@30a20000/sn65dsi84@2c: could not find display-timings node
+[    3.598985] /soc@0/bus@30800000/i2c@30a20000/sn65dsi84@2c: no timings specified
+(...)
+```
+
+Ok then, all `sn65dsi84` errors were fixed, it was a matter of devicetree
+configuration. But still no video on the panel.
+
+Several checks were done to check every part of the system separately:
+
+1) Checking `sn65dsi84` with LVDS panel proved correct driver and hardware
+configuration by showing some video output;
+2) Checking communication via I2C with `it6263` proved correct connection (some
+register write/read operations were done);
+3) Checking the HDMI panel proved correct connection and functionality by
+showing video output connected to another platform.
+
+Still not a clue where the problem is.
```

Comment:
Did you use Grammarly? I'm not sure if it would improve this and some other sentences.

Reply:
I did use Grammarly. It mostly argues about passive voice I used here to stay with impersonal form as much as possible. But you are right, this one can be rephrased. Fixed here 92b439ca1c263d38890f86404b8af0bd8834c42d .